modules/references/client/api/references.api.js

```diff
@@ -60,16 +60,21 @@ export async function create(reference) {
  *
  * @param {string} userFrom - id of user who gave the reference
  * @param {string} userTo - id of user who received the reference
+ * @params {bool} includeReplies - true if replies to the references need to be fetched
  * @returns Promise<Reference[]> - array of the found references
  */
-export async function read({ userFrom, userTo }) {
+export async function read({ userFrom, userTo, includeReplies }) {
   const params = {};
   if (userFrom) {
     params.userFrom = userFrom;
   }
   if (userTo) {
     params.userTo = userTo;
   }
+  if (includeReplies) {
```

Comment:
Any situation practically where we wouldn't want replies? If not, we could just always include them?

Reply:
I guess there might be potentially a situation where we wouldn't display replies. But even in this case, I can imagine that pre-loading them could be useful.  So, yes, I agree, let's always fetch replies. I think I got blinded by this code where we check if the user already shared an experience:

      const reference = await api.references.read({
        userFrom: userFrom._id,
        userTo: userTo._id,
        includeReplies: false,
      });

      if (reference.length === 1) {
        setIsDuplicate(true);
      }

This is the only case now where we pass `includeReplies=false`, but I think this logic just should be rewritten

Reply:
Yeah — additionally `userFrom _id, userTo: _id`  feels a very explicit condition already and it shouldn't need to include the "reply" with that condition.

For the purposes of the current UI, it even makes sense to just return boolean yes/no but it's also nice to have the whole experience object so that we can include some of it in the UI.

Reply:
The only thing I like about having `includeReplies` explicitly set to true is that it makes more clear why replies are returned: `/api/references?userTo=user1&includeReplies=true`.  Alternatively, we should change the route to `/api/referencePairs?user=user1` or something like this

Reply:
Btw, I think we don't really need `userFrom` param, we just need some boolean param like `onlyFromAuthUser` to check if there are already pending or published references from the signed in user:

https://github.com/Trustroots/trustroots/pull/1860/files#diff-7b97d7f1b923f70a20dd2d719b6b11324e2a13f048872f41fd1ca4e6b2d86bf3

Reply:
Fair point! Looks like we already use authenticated user's ID here: 

https://github.com/Trustroots/trustroots/blob/2c11ed4e7e59be349ba7719ee0510e5b7165a619/modules/references/server/controllers/reference.server.controller.js#L159

That `checkDuplicate()` method is currently used just inside `create()`, but perhaps it should be it's own API endpoint used just for checking this? That way the API for listing references doesn't need to be concerned with any of this?

Reply:
@simison What do you think about doing it like this: #1910

Reply:
@hlibbabii I like it! I'm not sure how soon I get to review but I'll try prioritize this.

Reply:
Great! After we merge that one, I'll remove `fromTo` and this PR will be ready to be reviewed